Sources/MessagingInApp/Views/UIKitInline.swift

```diff
@@ -113,66 +99,41 @@ public class InAppMessageView: UIView {
-            // We are already displaying or rendering a messsage. Do not show another message until the current message is closed.
-            // The main reason for this is when a message is tracked as "opened", the Gist backend will not return this message on the next fetch call.
-            // We want to coninue showing a message even if the fetch no longer returns the message and the message is currently visible.
+    public func onMessageRendered(width: CGFloat, height: CGFloat) {
+        // animate height change
+        guard let inAppMessageView = inAppMessageView else {
             return
         }
 
-        if let messageAvailableToDisplay {
-            displayInAppMessage(messageAvailableToDisplay)
-        } else {
-            dismissInAppMessage()
+        inAppMessageView.isHidden = false
+        animateHeight(to: height)
+
+        if let messageRenderingLoadingView = messageRenderingLoadingView {
+            animateFadeInOutInlineView(fromView: messageRenderingLoadingView, toView: inAppMessageView) {
+                messageRenderingLoadingView.removeFromSuperview()
+            }
         }
     }
 
-    // Function to check if a message has been previously shown
-    func hasBeenPreviouslyShown(_ message: Message) -> Bool {
-        previouslyShownMessages.contains { $0.id == message.id }
+    public func onNoMessageToDisplay() {
+        animateHeight(to: 0)
     }
 
-    private func displayInAppMessage(_ message: Message) {
-        // If this function gets called a lot in a short amount of time (eventbus triggers multiple events), the display animation does not look as expected.
-        // To fix this, exit early if display has already been triggered.
-        if let currentlyShownMessage = inlineMessageManager?.currentMessage, currentlyShownMessage.id == message.id {
-            return // already showing this message or in the process of showing it.
+    public func onInlineButtonAction(message: Message, currentRoute: String, action: String, name: String) -> Bool {
+        // If delegate is not set then call the global `messageActionTaken` method
+        guard let onActionDelegate = onActionDelegate else {
+            return false
         }
+        onActionDelegate.onActionClick(message: InAppMessage(gistMessage: message), actionValue: action, actionName: name)
+        return true
+    }
 
-        // If a different message is currently being shown, we want to replace the currently shown message with new message.
-        if isRenderingOrDisplayingAMessage {
-            showLoadingView {
-                // After animation is over, cleanup resources since we no longer need to show the previous message.
-                self.stopShowingMessageAndCleanup()
-                self.beginShowing(message: message)
-            }
-        } else {
-            beginShowing(message: message)
+    public func willChangeMessage(newTemplateId: String, onComplete: @escaping () -> Void) {
```

Comment:
`willChangeMessage` is called when the "show another message" action buttons are clicked. This function is called while the new message is being rendered.

Reply:
I think it would be helpful to add a comment on this method since it is `willChangeMessage` and will be called on "show another message" button click.

Reply:
Happy to add a comment. It would be helpful if you could provide a suggestion for the comment as I am not sure what questions you have about it. 

The `will_` naming convention is based on popular iOS APIs where that naming convention is used. The inline will will change the message that's shown when "show another message" button is clicked.